src/gsy_e/models/strategy/strategy_profile.py

```diff
@@ -128,7 +128,10 @@ def read_or_rotate_profiles(self, reconfigure=False):
             profile = self.profile
 
```

Comment:
It needs to be checked whether it violates the previous if clauses, however the solution can be similar to this, so that the profile variable is overridden if we need to rotate a profile that originates from a CSV file: 

```suggestion
if isinstance(self.input_profile, str):
    profile = self.input_profile
```

Reply:
` if isinstance(self.input_profile, str):` will always we True for all the market slot in the simulation which will result in calling the read_arbitrary_profile inside the ProfilesHandler on every market slot, which is not what we want.